data_pipeline/utils/data_store/bq_data_service.py

```diff
@@ -349,3 +351,70 @@ def load_new_data_in_temp_table_to_actual_table(
     query_job.result()  # Wait for the job to complete.
 
     LOGGER.info("Loaded table %s", table_id)
+
+
+def get_distinct_values_from_bq(
+            project_name: str,
+            dataset_name: str,
+            column_name: str,
+            table_name_source: str,
+            table_name_for_exclusion: str = None
+        ) -> pd.DataFrame:
+
+    sql = """
+        SELECT DISTINCT {{ column_name }}
+        FROM  `{{ project_name }}.{{ dataset_name }}.{{ table_name_source }}`
+        {% if table_name_for_exclusion %}
+        WHERE {{ column_name }} NOT IN
+            (
+                SELECT {{ column_name }}
+                FROM `{{ project_name }}.{{ dataset_name }}.{{ table_name_for_exclusion }}`
+            )
+        {% endif %}
+    """
+    params = {
+        'project_name': project_name,
+        'dataset_name': dataset_name,
+        'column_name': column_name,
+        'table_name_source': table_name_source,
+        'table_name_for_exclusion': table_name_for_exclusion
+        }
+
+    jin = JinjaSql(param_style='pyformat')
+    query, bind_params = jin.prepare_query(sql, params)
+    client = get_bq_client(project=project_name)
+    query_job = client.query(query % bind_params)  # Make an API request.
+    results = query_job.result()  # Waits for query to finish
+
+    return pd.concat([pd.DataFrame([row.column]) for row in results], ignore_index=True)
+
+
+def get_max_value_from_bq_table(
+            project_name: str,
+            dataset_name: str,
+            column_name: str,
+            table_name: str,
+        ) -> str:
+
+    sql = """
+        SELECT
+        MAX({{ column_name }}) AS start_id
+        FROM `{{ project_name }}.{{ dataset_name }}.{{ table_name }}`
+    """
+    params = {
+        'project_name': project_name,
+        'dataset_name': dataset_name,
+        'table_name': table_name,
+        'column_name': column_name
+        }
+
+    jin = JinjaSql(param_style='pyformat')
+    query, bind_params = jin.prepare_query(sql, params)
+    client = get_bq_client(project=project_name)
+    query_job = client.query(query % bind_params)  # Make an API request.
+    results = query_job.result()  # Waits for query to finish
+
+    for row in results:
```

Comment:
minor: now that this is "share", I would also suggest to just improve it very slightly. e.g. in this case the result will be exactly one row. So looping to me makes it a bit more difficult to read. I would also not convert it to a string. If that was desired, the caller could easily do it.

e.g.:

```python
return list(results)[0].value
```